[#182998] Avisar usuário quando o microfone não estiver autorizado pe…

BrunoFAires opened the pull request:
…lo navegador

- Passagem das functions de callback para o ponto exato onde ocorre a lógica de qual "modal" exibir.

BrunoFAires committed:
[#182998] Avisar usuário quando o microfone não estiver autorizado pelo navegador

- Passagem das functions de callback para o ponto exato onde ocorre a lógica de qual "modal" exibir.

gitTecsinapse commented:
https://tecsinapse.tpondemand.com/entity/182998

gabrielbutka commented on src/components/Chat/Maximized/InputComposer/InputComposer.js:
```diff
@@ -122,7 +124,32 @@ export const InputComposer = ({
+
+  const onClick = () => {
+    navigator.permissions.query({ name: 'microphone' }).then(r => {
+      console.log(r);
```
retirar console.log

gabrielbutka reviewed (approved):
só deixei uma observação para retirar um console.log

BrunoFAires committed:
Revert "[#182998] Avisar usuário quando o microfone não estiver autorizado pelo navegador"

This reverts commit d253f84012522eaec26642d1ffdef0a86472b3c3.

BrunoFAires committed:
[#182998] Avisar usuário quando o microfone não estiver autorizado pelo navegador

- Criação do arquivo de constantes RESOURCES
- Criação de um arquivo utils
- Migração da lógica em relação à abertura do modal para o InputComposer.

BrunoFAires commented:
Foi possível, como uso do componente Dialog, trazer toda a lógica de abertura do modal que solicita a permissão do microfone para a lib do Chat, mas especificamente no InputComposer. Dessa forma, a verificação da permissão, e consequentemente solicitação, fica do lado do Chat e não do projeto que importa ele. Sendo assim, acredito que essa questão fica centralizada do lado mais "correto" possível. @gabrielbutka

gabrielbutka commented on src/components/Chat/Maximized/InputComposer/InputComposer.js:
```diff
@@ -122,14 +133,80 @@ export const InputComposer = ({
+
+    return (
+      <DialogContentText>
+        Para gravar uma mensagem de voz cliquem em &quot;Permitir&quot;, logo
```
'clique' ao invés de 'cliquem'

BrunoFAires committed:
[#182998] Avisar usuário quando o microfone não estiver autorizado pelo navegador

- Aplicação CR

BrunoFAires committed:
Merge branch 'master' into feature/tp-182998-solicitar-acesso-ao-microfone-para-usuario
